src/Sarif.Multitool.FunctionalTests/TestData/UriBaseIdRequiresRelativeUri/AbsoluteUriInInvocationAttachment.sarif

```diff
@@ -0,0 +1,24 @@
+{
+  "$schema": "http://json.schemastore.org/sarif-2.0.0",
+  "version": "2.0.0",
+  "runs": [
+    {
+      "tool": {
+        "name": "CodeScanner"
```

Comment:
it would be great if a test file had some marker to explicitly denote all expected failures. the property bag could be utilized. what's your general approach here for making this explicit? where are you putting positive test cases? sorry i am so late to this code, we could get on the phone for a chat about it if you'd like to fill me in that way.

Reply:
For each rule I have a separate negative test case for each possible error, and then a single positive case showing all the correct content. So for example `TestData\UriBaseIdRequiresRelativeUri\UrisAreRelative.sarif` is the positive test case. Likewise, `TestData\UrisMustBeValid\ValidUris.sarif` is the positive test case for that rule.

I'd be willing to discuss adding a documenting `message` property somewhere in each test file, but I think if I keep them small enough (as I mentioned in response to another comment) they will be understandable.

---
In reply to: [210452808](https://github.com/Microsoft/sarif-sdk/pull/1018#discussion_r210452808) [](ancestors = 210452808)